### PR TITLE
[TA1304] unit tests for negative cases in target for network connections 

### PR DESCRIPTION
**Pull request by mynktl**
Changes : 
- Added error injection support in `istgt_integration` test utility. 
    As of now, error injection is added for below scenarios : 
       1. Closing management or data connection from replica
       2. Sending multiple responses for a single management command from replica
       3. Sending errored response for IOs from replica

- Code fix for the errors reported by error injection feature.

### CODE REVIEW

File: src/replication.c [hunk 1]
```diff
@@ -1479,6 +1478,10 @@ accept_mgmt_conns(int epfd, int sfd)
 	int mgmt_fd;
 	mgmt_event_t *mevent1, *mevent2;
 
+/*
```
Comment (vishnuitta) [inline src/replication.c]: this is not required

File: src/replication.c [hunk 2]
```diff
@@ -2253,6 +2259,8 @@ handle_mgmt_conn_error(replica_t *r, int sfd, struct epoll_event *events, int ev
 				if (r_ev == r) {
 					TAILQ_REMOVE(&r->spec->rwaitq, r, r_waitnext);
 					r->conn_closed++;
+					if (r->io_resp_hdr)
```
Comment (vishnuitta) [inline src/replication.c]: set io_resp_hdr to NULL here, and also, in handle_data_conn_error() (with lock)

File: src/replication.c (hunk 1 above)
Comment (mynktl) [reply src/replication.c]: ok

File: src/replication.c (hunk 2 above)
Comment (mynktl) [reply src/replication.c]: ok

**Comment by mynktl** [comment]
Hi @vishnuitta
I have addressed all review comments given by you. I have also enabled error injection for IOs and added comments for replica structure variables.

**Review by vishnuitta — APPROVED** [review]
changes are good.

**Pull request merged**